### PR TITLE
[16.0][MIG] stock_inventory_lockdown: Migration to 16.0

### PR DESCRIPTION
**Pull request by ArnauCForgeFlow**
Standard Migration to 16.0
Also fwport of (from https://github.com/OCA/stock-logistics-warehouse/pull/2155):
- https://github.com/OCA/stock-logistics-warehouse/commit/cc9166ece0720b66bc18be3745c01325350664f2
- https://github.com/OCA/stock-logistics-warehouse/commit/c243f874727938c170ca136c27b4087ba1f449d4

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
Functional review :+1:

File: stock_inventory_lockdown/tests/test_stock_inventory_lockdown.py [hunk 1]
```diff
@@ -98,23 +100,27 @@ def test_inventory(self):
...
+                "stock_quant_ids": line,
             }
         )
-        self.assertTrue(inventory_subloc.line_ids)
-        inventory_subloc.action_validate()
+        inventory_subloc.action_state_to_in_progress()
+
+        self.assertTrue(inventory_subloc.stock_quant_ids)
+
+        # for line in inventory_subloc.stock_quant_ids:
```
Comment (rousseldenis) [inline stock_inventory_lockdown/tests/test_stock_inventory_lockdown.py]: Could you remove this ?

**Review by JoanSForgeFlow — APPROVED** [review]
LGTM!

File: stock_inventory_lockdown/models/stock_location.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def _check_inventory_location_id(self):
+        vals = set(self.ids) | set(self.mapped("location_id").ids)
+        location_inventory_open_ids = self.env[
+            "stock.inventory"
+        ]._get_locations_open_inventories(vals)
+        if location_inventory_open_ids:
+            raise ValidationError(_("An inventory is being conducted at this location"))
+
+    def unlink(self):
+        location_inventory_open_ids = (
```
Comment (rousseldenis) [inline stock_inventory_lockdown/models/stock_location.py:21]: Maybe increasing coverage by testing `unlink()`

File: stock_inventory_lockdown/models/stock_move_line.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            ):
+                location_names = locked_location_ids.mapped("complete_name")
+                raise ValidationError(
+                    _(
+                        "Inventory adjusment underway at "
+                        "the following location(s):\n- %s\n"
+                        "Moving products to or from these locations is "
+                        "not allowed until the inventory adjustment is complete."
+                    )
+                    % "\n - ".join(location_names)
```
Comment (rousseldenis) [inline stock_inventory_lockdown/models/stock_move_line.py]: Please use translation function arguments instead.

File: stock_inventory_lockdown/tests/test_stock_inventory_lockdown.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+# © 2016 Numérigraphe SARL
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.exceptions import ValidationError
+
+from odoo.addons.stock.tests.common import TestStockCommon
+
+
+class StockInventoryLocationTest(TestStockCommon):
+    def setUp(self):
```
Comment (rousseldenis) [inline stock_inventory_lockdown/tests/test_stock_inventory_lockdown.py]: Switch to classmethod

**Comment by ArnauCForgeFlow** [comment]
@rousseldenis done!

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**